### PR TITLE
Add cover platform

### PR DESCRIPTION
**Pull request by engrbm87**
## Proposed changed

Add cover platform to UI. The settings for the relay and state ports are located under the options menu.

### CODE REVIEW

File: custom_components/rpi_gpio/cover.py [hunk 1]
```diff
@@ -51,87 +59,116 @@
...
+        """Return if state port should be inverted."""
+        return self.entry.options.get(CONF_INVERT_STATE, DEFAULT_INVERT_STATE)
 
-    def _trigger(self):
+    @property
+    def relay_time(self) -> float:
+        """Return the relay turn on duration."""
+        return self.entry.options.get(CONF_RELAY_TIME, DEFAULT_RELAY_TIME)
+
+    async def async_update(self) -> None:
```
Comment (thecode) [inline custom_components/rpi_gpio/cover.py:129]: I think we need to call this first time as a last step in `__init__`

**Comment by thecode** [comment]
I received this error when trying to invert the `RELAY` state:
```
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 281, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/rpi_gpio/cover.py", line 87, in async_setup_entry
    await hass.async_add_executor_job(rpi_gpio.setup_port, entry)
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/rpi_gpio/__init__.py", line 156, in setup_port
    GPIO.add_event_detect(
RuntimeError: Conflicting edge detection already enabled for this GPIO channel
```
I think we need to remove the old edge detect before making a new one.

Another problem is that due to edge detect the initial state seems wrong (left a comment for a suggested solution)

File: custom_components/rpi_gpio/cover.py (hunk 1 above)
Comment (engrbm87) [reply custom_components/rpi_gpio/cover.py:129]: This is already called when setting it to True
https://github.com/thecode/ha-rpi_gpio/blob/0d51d93541e1e7bb2fb21fba5beefa3f95199511/custom_components/rpi_gpio/cover.py#L89

**Comment by thecode** [comment]
I am trying to setup options using the `CONFIGURE` and only the denounce time can be updated, all others return to defaults when you open the dialog again:
image

**Comment by engrbm87** [comment]
> I am trying to setup options using the `CONFIGURE` and only the denounce time can be updated, all others return to defaults when you open the dialog again: image

Sorry about that, should double check and copying code🤦‍♂️. Fixed in last commit.

**Comment by ScriptedBrain** [comment]
I've tested your integration, and found the following problem.

I have an apartments porch door, which I open via a relay on the gpio of the Pi I am running home assistant on. I've configured the relay where both input and relay pins are inverted (So give power to the relay when I want to open the door). That functions well. But whenever home assistant restarts, the relay defaults to the ON state. Meaning the door will be open, until I command it to open. Then it puts the relay to the OFF state. I have this with the relay pin inverted and not inverted. It always defaults to ON state on start.

**Comment by ScriptedBrain** [comment]
> Have you tested the changes in this pull request?
Yes, I pulled the latest code from this PR. Even did it again just to be sure. But every time when the GPIO integration starts it pulls the relay pin high until cover open/close service is called.

**Comment by thecode** [comment]
@engrbm87 I have also noticed that, this is why I thought we have problem with the update, we need to address this

**Comment by engrbm87** [comment]
I am looking at the current code for the cover and I found that we are writing 1 if the invert_relay option is True. This means we are assume that the output pin should be HIGH by default and we invert it to make LOW.
If this is how it should be and we want to keep it this way then I just need to change the initialization of the output pin to make it LOW if inverted and High by default.
@thecode , can you please advise on this.

**Pull request closed without merging**